packages/yarnpkg-core/sources/MessageName.ts

```diff
@@ -72,6 +74,25 @@ export enum MessageName {
   INVALID_CONFIGURATION_VALUE = 67
 }
 
-export function stringifyMessageName(name: MessageName | number): string {
+export class PluginMessageName {
+  constructor(readonly pluginName: string, readonly messageKey: string, readonly messageId: number, readonly messageName: string) {
+  }
+}
+
+export function createPluginMessageName<M extends {[messageName: string]: number}>(pluginName: string, messages: M): Record<keyof M, PluginMessageName> {
```

Comment:
I think it would be best to implement this in a separate diff; the logfilter is ready to ship, but there are additional discussions about plugin messages (for example is it necessary to include the pluginName here, since the plugin names are already part of the generated bundle?)

Reply:
Sure, can do! I'd still like to remove the message text filtering though (see https://github.com/yarnpkg/berry/pull/1907#discussion_r516145183) so this would mean that as it stands the filtering would only support yarn's own messages (with support for plugins coming once the PluginMessageName is added)